server/server.go

```diff
@@ -236,15 +236,15 @@ func (server *HTTPServer) voteOnRule(writer http.ResponseWriter, request *http.R
 	if err != nil {
 		const message = "Unable to get user id"
 		log.Error().Err(err).Msg(message)
-		responses.Send(http.StatusInternalServerError, writer, message)
+		responses.SendError(writer, message)
```

Comment:
it seems to be more like a server error, because we wouldn't be authorized if user didn't send auth token or sent the wrong one

Reply:
how can this happen then? when the user doesn't supply the credentials, it returns Auth error, before reaching the func, right?

Reply:
and if auth were to be disabled, isn't it a User error when he doesn't supply the identity?

Reply:
only if server put wrong type in authentication middleware https://github.com/RedHatInsights/insights-results-aggregator/blob/c8cec81e336323fef16fe6a86fa51faae6703bc7/server/auth.go#L130

Reply:
@Sergey1011010 so wait, can this happen entirely by user fault? Or is it really a combination of server conf that would cause that to happen even if the request is correct?

I mean I'm trying to understand if it this error can be fixed by the user (eg. by sending the correct identity), then it should be 4xx, but if not then i of course agree with you

Reply:
No, the user has nothing to do with it, it can only happen if someone modifies this line of code https://github.com/RedHatInsights/insights-results-aggregator/blob/c8cec81e336323fef16fe6a86fa51faae6703bc7/server/auth.go#L122 and puts different type there

Reply:
ok, I changed it back